### PR TITLE
Message PopUp: Prototype of mobile web scoring flow

### PR DESCRIPTION
**Pull request by kevinrobinson**
This is a functional prototype of a scoring flow for Message PopUp targeting mobile web.  It includes the work for this feature along with some related refactorings.

#### List of questions
This shows the list of questions that need to be scored.  Each question may have multiple responses from different students.  The icons are placeholder now, but could mirror what Joyce and Rupal come up with for different groups of competencies.
screen shot 2016-06-29 at 9 46 16 am

#### Scoring responses for a question
Tapping into a question shows the context about that question, including the question itself, card for the student, and any examples and non-examples.  These can be toggled open or closed.
screen shot 2016-06-29 at 12 12 08 pm

#### Swiping to scope
The intent of this prototype is for scorers to not feel overwhelmed and burdened by the amount of work, and to feel like they can tick down responses that need scoring fairly quickly.  Other learning experiences will certainly require more contemplation from faculty and evaluators, but this is an experiment with a quicker feedback format for this particular experience.

Swiping right scores it as "competency demonstrated," dismissing it and recording the score.
screen shot 2016-06-29 at 12 37 51 pm

And swiping left brings up a feedback note:
screen shot 2016-06-29 at 12 23 03 pm

#### Related changes and refactorings
- Introduces react-swipeable-views and react-motion for prototyping the swipe interaction
- Adds a new database table for holding evaluations
- Factors out API calls into an `api` module
- Adds in styling prop to `StudentCard` to support re-use in the scoring page
- Adds in question ids to the question data, and a workaround function for historical responses that don't have these ids

### CODE REVIEW

**Comment by kevinrobinson** [comment]
@david-rosales This is a larger pull request since it's trying to prototype the interactions.  I probably should have split it into smaller pieces to start, which definitely makes it easier to review!

Can you review when you have a minute?  And feel free to ask any questions on the review too! :)

File: ui/src/message_popup/transformations.jsx [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    const learningObjective = _.find(learningObjectives, { id: question.learningObjectiveId });
+    if (learningObjective.competencyGroup !== competencyGroup) return null;
+    return {
+      ...question,
+      competencyGroup: learningObjective.competencyGroup
+    };
+  }));
+
+  return _.shuffle(withStudents(withCompetencyGroups));
+}
```
Comment (david-rosales) [inline ui/src/message_popup/transformations.jsx]: I am about to make a commit to my open pull request that moved these to the inside of `instructions_card.jsx` but I think that this is a much better idea so we should just make sure to change that when merging whichever one is merged last.

**Comment by david-rosales** [comment]
@kevinrobinson Looks good!

File: ui/src/message_popup/transformations.jsx (hunk 1 above)
Comment (kevinrobinson) [inline ui/src/message_popup/transformations.jsx]: @david-rosales sounds great! 👍

**Pull request merged**